addons/website_sale_delivery/controllers/main.py

```diff
@@ -193,17 +193,24 @@ def express_checkout_process_shipping_address(self, partial_shipping_address):
                     name=_('Anonymous express checkout partner for order %s', order_sudo.name),
                 )
 
-        # Returns the list of develivery carrier available for the sale order.
-        return sorted([{
-            'id': carrier.id,
-            'name': carrier.name,
-            'description': carrier.website_description,
-            'minorAmount': payment_utils.to_minor_currency_units(
-                WebsiteSaleDelivery._get_rate(carrier, order_sudo, is_express_checkout_flow=True)['price'],
-                order_sudo.currency_id,
-            ),
-        } for carrier in order_sudo._get_delivery_methods()],
-        key=lambda carrier: carrier['minorAmount'])
+        # Returns the list of delivery carrier available for the sale order.
+        return sorted(
+            [{
+                'id': carrier.id,
+                'name': carrier.name,
+                'description': carrier.website_description,
+                'minorAmount': payment_utils.to_minor_currency_units(price, order_sudo.currency_id),
+            } for carrier, price in self.get_carriers_express_checkout(order_sudo).items()
+            ], key=lambda carrier: carrier['minorAmount']
+        )
+
+    def get_carriers_express_checkout(self, order):
```

Comment:
Probably better to make it private
```suggestion
    def _get_carriers_express_checkout(self, order):
```

Reply:
yes, I am not sure if we want the second commit for 16.0 version, if yes then I will squash them. Because nobody yet complained about fedex/sencloud use locations in the express checkout before.